src/Entity/Transformer.php

```diff
@@ -40,13 +47,15 @@ final class Transformer implements TransformerInterface
     private array $children = [];
 
     /**
-     * @param ManagerInterface<PageBlockInterface> $blockManager
+     * @param ManagerInterface<PageBlockInterface>                          $blockManager
+     * @param SerializerInterface&NormalizerInterface&DenormalizerInterface $serializer
      */
     public function __construct(
         private SnapshotManagerInterface $snapshotManager,
         private PageManagerInterface $pageManager,
         private ManagerInterface $blockManager,
-        private ManagerRegistry $registry
+        private ManagerRegistry $registry,
+        private ?SerializerInterface $serializer = null
     ) {
     }
```

Comment:
You need to add 
```
// NEXT_MAJOR: Remove remove null support
if (null === $this->serializer) {
    @trigger_error(sprintf(
                'Not passing an instance of %s as 5th parameter is deprecated since version 4.x and will be removed in 5.0.',
                SerializerInterface::class,
                __METHOD__
            ), \E_USER_DEPRECATED);
}
```

Reply:
i need to see what to do with the trigger error for my TransformerTest without Serializer